algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/treatment/CurrentlyGetsChemoradiotherapyWithSpecificChemotherapyTypeAndMinimumCycles.kt

```diff
@@ -6,25 +6,62 @@ import com.hartwig.actin.datamodel.PatientRecord
 
-        val matchingTreatments = record.oncologicalHistory.filter {
-            it.categories().containsAll(setOf(TreatmentCategory.CHEMOTHERAPY, TreatmentCategory.RADIOTHERAPY)) &&
-                    it.isOfType(type) == true &&
-                    it.treatmentHistoryDetails?.cycles != null &&
-                    it.treatmentHistoryDetails?.cycles!! > minCycles
+    override fun evaluate(record: PatientRecord): Evaluation {
+        val treatmentMatches = record.oncologicalHistory.groupBy {
+            when {
+                (it.categories().containsAll(setOf(TreatmentCategory.CHEMOTHERAPY, TreatmentCategory.RADIOTHERAPY)) &&
+                        it.isOfType(type) == true &&
+                        enoughCyclesAndOngoingTreatment(it.treatmentHistoryDetails) == true) -> {
+                    true
+                }
+                !(it.categories().containsAll(setOf(TreatmentCategory.CHEMOTHERAPY, TreatmentCategory.RADIOTHERAPY)) ||
+                        it.isOfType(type) == false ||
+                        enoughCyclesAndOngoingTreatment(it.treatmentHistoryDetails) == false) -> {
+                    false
+                }
+                else -> {
+                    null
+                }
+            }
         }
 
         return when {
-            matchingTreatments.isNotEmpty() -> EvaluationFactory.pass("Patient is currently getting chemoradiotherapy with $type chemotherapy and at least $minCycles cycles")
+            treatmentMatches.isEmpty() -> EvaluationFactory.fail("No treatments received")
+            true in treatmentMatches -> EvaluationFactory.pass("Patient is currently getting chemoradiotherapy with $type chemotherapy and at least $minCycles cycles")
+            null in treatmentMatches -> EvaluationFactory.undetermined("Undetermined if patient is currently getting chemoradiotherapy with $type chemotherapy and at least $minCycles cycles")
             else -> EvaluationFactory.fail("No chemo radio therapy with $type with at least $minCycles is currently received")
-            //else -> EvaluationFactory.undetermined("Undetermined if patient is currently getting chemoradiotherapy with $type chemotherapy and at least $minCycles cycles" )
+        }
+    }
+
+    fun enoughCyclesAndOngoingTreatment(treatmentHistoryDetails: TreatmentHistoryDetails?): Boolean?{
+        if (treatmentHistoryDetails == null)
+            return null
+
+        val cycles = treatmentHistoryDetails.cycles
+        if(cycles == null)
+            return null
+        else if(cycles < minCycles)
+            return false
+
+        val stopYear = treatmentHistoryDetails.stopYear
+        val stopMonth = treatmentHistoryDetails.stopMonth
+
+        return when {
+            stopYear == null -> null
+            now.year < stopYear -> true
+            now.year > stopYear -> false
+            stopMonth == null -> null
+            now.monthValue <= stopMonth -> true
+            else -> false
         }
```

Comment:
I try to avoid putting multiple `return` statements in a single code block because it can make code harder to follow. It's not that different from using `break` or even `goto`. There are also some useful date comparison functions you can use here:
```suggestion
        return treatmentHistoryDetails?.cycles?.let { cycles ->
            val appearsOngoing = with (treatmentHistoryDetails) {
                DateComparison.isAfterDate(referenceDate, stopYear, stopMonth) != false
            }
            cycles >= minCycles && appearsOngoing
        }
```